pulp_ansible/app/galaxy/v3/views.py

```diff
@@ -295,15 +304,29 @@ def update(self, request, *args, **kwargs):
         serializer = self.get_serializer(collection, data=request.data, partial=True)
         serializer.is_valid(raise_exception=True)
         deprecated_value = request.data.get("deprecated")
+        add_content_units = []
+        remove_content_units = []
+
+        deprecation, created = AnsibleCollectionDeprecated.objects.get_or_create(
```

Comment:
@daviddavis with the new orphan cleanup behavior does anything additional need to happen here like an additional touch() call?

Reply:
Yes, it's ugly but I think we'll need something like this:

```python
deprecation, created = AnsibleCollectionDeprecated.objects.get_or_create(
    namespace=collection.namespace, name=collection.name
)
if not created:
    try:
        deprecation.touch()
    except DatabaseError:
        # deprecation has since been removed. try to create it
        deprecation = AnsibleCollectionDeprecated.objects(namespace=collection.namespace, name=collection.name)